borsh-schema-derive-internal/src/helpers.rs

```diff
@@ -8,6 +8,19 @@ pub fn contains_skip(attrs: &[Attribute]) -> bool {
         .any(|attr| attr.path().to_token_stream().to_string().as_str() == "borsh_skip")
 }
 
+pub fn filter_skip(attrs: &[Attribute]) -> Vec<Attribute> {
+    attrs
+        .iter()
+        .filter_map(|attr| {
+            if attr.path().to_token_stream().to_string().as_str() == "borsh_skip" {
+                Some(attr.clone())
+            } else {
+                None
+            }
+        })
```

Comment:
nit: `filter` + `cloned` looks easier than `filter_map`:

```suggestion
pub fn filter_skip(attrs: &[Attribute]) -> Vec<&Attribute> {
    attrs
        .iter()
        .filter(|attr| attr.path().to_token_stream().to_string() == "borsh_skip")
        .cloned()
```